### PR TITLE
[risk=low] [DB-672] Filter survey sub question results by path

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
image

Survey question results are repeated. 
Filtering them by path.

Eg: For a question Why did your periods stop branched from Answer Yes, I have no menstrual periods  also has answers from Why did your periods stop branched from Answer Yes, but from induced harmones.

Filtering by path to avoid duplication

### CODE REVIEW

File: public-api/db-cdr/csv/achilles_analysis.csv [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
-3106,"Current Age histogram","concept_id","current_age_decile","domain_id",NULL,NULL,"column","counts"
-3107,"Gender Identity","concept_id","gender_identity_concept_id","domain_id",NULL,NULL,"pie","counts"
-3108,"Race Ethnicity","concept_id","race_concept_id","domain_id",NULL,NULL,"column","counts"
-3110,"Response Distribution","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","question_order","column","counts"
-3111,"Biological Sex","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","gender_concept_id","column","counts"
-3112,"Age at occurrence","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","age_decile","column","counts"
-3113,"Gender Identity","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","gender_identity_concept_id","column","counts"
-3114,"Race ethnicity","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","race_ethnicity_concept_id","column","counts"
+analysis_id,"analysis_name","stratum_1_name","stratum_2_name","stratum_3_name","stratum_4_name","stratum_5_name","stratum_6_name","chart_type","data_type"
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/csv/achilles_analysis.csv:1]: Data changes needed for local development

File: public-api/db-cdr/generate-cdr/bq-schemas/achilles_analysis.json [hunk 2]
```diff
@@ -28,7 +28,11 @@
 	{
 	"name": "stratum_5_name", 
 	"type": "STRING"
-	}, 
+	},
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/bq-schemas/achilles_analysis.json:31]: Schema change to add what is the value in stratum 6.

File: public-api/db-cdr/generate-cdr/csv/achilles_analysis.csv [hunk 3]
```diff
@@ -1,36 +1,32 @@
...
-3108,"Race Ethnicity","concept_id","race_concept_id","domain_id",NULL,NULL,"column","counts"
-3110,"Response Distribution","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","question_order","column","counts"
-3111,"Sex Assigned at Birth","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","gender_concept_id","column","counts"
-3112,"Age at occurrence","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","age_decile","column","counts"
-3113,"Gender Identity","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","gender_identity_concept_id","column","counts"
-3114,"Race ethnicity","survey_concept_id","question_concept_id","answer_concept_id","answer_value_string","race_ethnicity_concept_id","column","counts"
-100,"Physical Measurements Concept Group Metadata","group_title","group_name","concept_1","concept_2","concept_3","",""
+analysis_id,"analysis_name","stratum_1_name","stratum_2_name","stratum_3_name","stratum_4_name","stratum_5_name","stratum_6_name","chart_type","data_type"
```
Comment (SrushtiGangireddy) [inline public-api/db-cdr/generate-cdr/csv/achilles_analysis.csv:1]: Data changes needed for count generation.

File: public-api/src/main/java/org/pmiops/workbench/cdr/model/AchillesResult.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
     private String stratum3;
     private String stratum4;
     private String stratum5;
+    private String stratum6;
```
Comment (SrushtiGangireddy) [inline public-api/src/main/java/org/pmiops/workbench/cdr/model/AchillesResult.java:20]: Need stratum6 (As it contains survey question path in all the survey analyses).

File: public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java [hunk 5]
```diff
@@ -349,6 +349,7 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                             .stratum3(o.getStratum3())
                             .stratum4(o.getStratum4())
                             .stratum5(o.getStratum5())
+                            .stratum6(o.getStratum6())
```
Comment (SrushtiGangireddy) [inline public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java:352]: Added stratum6.

File: public-ui/src/app/views/survey-view/survey-view.component.ts [hunk 6]
```diff
@@ -142,6 +142,9 @@ export class SurveyViewComponent implements OnInit, OnDestroy {
                   }
                   subQuestion.graphToShow = GraphType.BiologicalSex;
                   subQuestion.selectedAnalysis = subQuestion.genderAnalysis;
+                  subQuestion.countAnalysis.surveyQuestionResults =
```
Comment (SrushtiGangireddy) [inline public-ui/src/app/views/survey-view/survey-view.component.ts:145]: Filter the results by the ones that only belong to this path.

**Review by mhigbyflowers — APPROVED** [review]
👍 thanks

**Pull request merged**